### DISCUSSION TITLE
User interactions on activities

### DISCUSSION THREAD

**Discussion by rmanjusha** [elsa-workflows/elsa-core#5582] [Q&A]
Does Elsa support user interactions on activity ? Please provide  a sample

**Comment by glime-ai** [comment]
Yes, Elsa does support user interactions on activities. User interactions can be facilitated through activities such as `HTTP Endpoint`, which can receive HTTP requests, or `ReceiveHttpRequest`, which waits for an incoming HTTP request at a specified path. Additionally, custom activities can be created to handle specific user interactions as needed.

For example, you could use the `HTTP Endpoint` activity to start a workflow with a form submission. When the form is submitted, the HTTP request triggers the workflow, and subsequent activities can process the form data.

Here's a simple example of a workflow that includes user interaction:

```csharp
var workflow = builder
    .StartWith<HttpEndpoint>(setup => setup
        .WithPath("/submit-form")
        .WithMethod(HttpMethods.Post))
    .Then<WriteHttpResponse>(setup => setup
        .WithContent(context => "Thank you for submitting the form!")
        .WithContentType("text/plain")
        .WithStatusCode(HttpStatusCode.OK));
```

In this example, the workflow starts with an `HttpEndpoint` activity that listens for POST requests on the path `/submit-form`. When a request is received, the workflow moves to the `WriteHttpResponse` activity, which sends a response back to the user.

Please note that the above code is a conceptual example to illustrate how user interactions can be handled in Elsa. The actual implementation may vary based on the version of Elsa you are using and the specific requirements of your application.

For more detailed examples and guides, you should refer to the official Elsa documentation and community resources.

This comment was generated by [Glime](https://glimelab.ai/?utm_source=dRbvTb7IhCSYiI&utm_medium=github).
